pkg/plugins/manager/loader/loader.go

```diff
@@ -25,41 +26,106 @@ var _ plugins.ErrorResolver = (*Loader)(nil)
-func ProvideService(cfg *config.Cfg, license plugins.Licensing, authorizer plugins.PluginLoaderAuthorizer,
-	pluginRegistry registry.Service, backendProvider plugins.BackendFactoryProvider, pluginFinder finder.Finder,
-	roleRegistry plugins.RoleRegistry, assetPath *assetpath.Service, signatureCalculator plugins.SignatureCalculator) *Loader {
-	return New(cfg, license, authorizer, pluginRegistry, backendProvider, process.NewManager(pluginRegistry),
-		roleRegistry, assetPath, pluginFinder, signatureCalculator)
+func ProvideService(cfg *config.Cfg, authorizer plugins.PluginLoaderAuthorizer,
+	pluginRegistry registry.Service, pluginFinder finder.Finder, assetPath *assetpath.Service,
+	signatureCalculator plugins.SignatureCalculator, hooksRegistry hooks.Registry, hooksRunner hooks.Runner,
+	pluginInitializer initializer.Initializer,
+
+	// TODO: hooks: Provided just for hooks side-effects, find a better way
+	_ process.Service,
+) *Loader {
+	return New(cfg, authorizer, pluginRegistry, assetPath, pluginFinder, signatureCalculator, pluginInitializer, hooksRegistry, hooksRunner)
 }
 
-func New(cfg *config.Cfg, license plugins.Licensing, authorizer plugins.PluginLoaderAuthorizer,
-	pluginRegistry registry.Service, backendProvider plugins.BackendFactoryProvider,
-	processManager process.Service, roleRegistry plugins.RoleRegistry,
-	assetPath *assetpath.Service, pluginFinder finder.Finder, signatureCalculator plugins.SignatureCalculator) *Loader {
+func New(cfg *config.Cfg, authorizer plugins.PluginLoaderAuthorizer,
+	pluginRegistry registry.Service,
+	assetPath *assetpath.Service, pluginFinder finder.Finder, signatureCalculator plugins.SignatureCalculator,
+	pluginInitializer initializer.Initializer,
+	hooksRegistry hooks.Registry, hooksRunner hooks.Runner) *Loader {
+	logger := log.New("plugin.loader")
+
+	// TODO: hooks: Move those to separate services
+	hooksRegistry.RegisterBeforeLoadHook(func(ctx context.Context, plugin *plugins.Plugin) error {
+		if plugin.IsApp() {
+			setDefaultNavURL(plugin)
+		}
+		return nil
+	})
+	hooksRegistry.RegisterBeforeLoadHook(func(ctx context.Context, plugin *plugins.Plugin) error {
+		if plugin.Parent != nil && plugin.Parent.IsApp() {
+			configureAppChildPlugin(plugin.Parent, plugin)
+		}
+		return nil
+	})
+
+	// This hook MUST run AFTER all other before init hooks
+	// TODO: hooks: The hook is registered here in Loader rather than Initialize to enforce it as the last one to run.
+	//		Find a better way to handle those cases as unfortunately this hooks depends on the ones above
+	//		(for validation etc). Or maybe do not make the Initializer a hook at all...
+	hooksRegistry.RegisterBeforeLoadHook(pluginInitializer.Initialize)
+
+	hooksRegistry.RegisterLoadHook(func(ctx context.Context, plugin *plugins.Plugin) error {
+		if !plugin.IsCorePlugin() && !plugin.IsBundledPlugin() {
+			metrics.SetPluginBuildInformation(plugin.ID, string(plugin.Type), plugin.Info.Version, string(plugin.Signature))
+		}
+		return nil
+	})
```

Comment:
This should be moved to a separate service (like PluginMetrics) in a follow-up PR